verification/curator-service/ui/src/components/LinelistTable.tsx

```diff
@@ -35,31 +46,54 @@ export default class LinelistTable extends React.Component<{}, TableState> {
         }
     }
 
+    handleChangePage = (event: unknown, newPage: number) => {
+        this.setState({ page: newPage });
+    };
+
+    handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+        this.setState({ rowsPerPage: parseInt(event.target.value, 10) });
+    };
+
     render() {
-        const { errorMessage, isLoaded, linelist } = this.state;
+        const { errorMessage, isLoaded, linelist, page, rowsPerPage } = this.state;
         if (errorMessage.length > 0) {
             return <div>Error: {errorMessage}</div>;
         } else if (!isLoaded) {
             return <div>Loading...</div>;
         }
         return (
-            <Table bordered hover striped>
-                <thead>
-                    <tr>
-                        <th>ID</th>
-                        <th>Date</th>
-                        <th>Outcome</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    {linelist.map(item => (
-                        <LinelistCaseRow
-                            case={item}
-                            key={item._id}
-                        />
-                    ))}
-                </tbody>
-            </Table>
+            <Paper>
+                <TableContainer>
+                    <Table aria-label="simple table">
+                        <TableHead>
+                            <TableRow style={{ background: '#cfd8dc' }}>
+                                <TableCell>ID</TableCell>
+                                <TableCell>Date</TableCell>
+                                <TableCell>Outcome</TableCell>
+                            </TableRow>
+                        </TableHead>
+                        <TableBody >
+                            {(rowsPerPage > 0
+                                ? linelist.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
```

Comment:
presumably at some point we'll push the pagination to the API layer and only fetch the rows in the current page?

Reply:
We'll definitely change this approach at some point. Something like: send an initial request on mount with just the first `rowsPerPage`, then an async request loading the next n or remainder. This "pagination" is mostly about the display, but will have to be extended when we get to that (if it's still around).